rio/ntriples/src/main/java/org/eclipse/rdf4j/rio/ntriples/NTriplesParser.java

```diff
@@ -470,22 +472,59 @@ else if (c != ':') {
 		if (c == -1) {
 			throwEOFException();
 		}
-		else if (!NTriplesUtil.isLetterOrNumber(c)) {
-			reportError("Expected a letter or number, found: " + new String(Character.toChars(c)),
+		else if (!NTriplesUtil.isLetterOrNumber(c) && !NTriplesUtil.isUnderscore(c)) {
+			reportError("Expected a letter or number or underscore, found: " + new String(Character.toChars(c)),
 					NTriplesParserSettings.FAIL_ON_NTRIPLES_INVALID_LINES);
 		}
 		name.append(Character.toChars(c));
 
 		// Read all following letter and numbers, they are part of the name
 		c = readCodePoint();
-		while (c != -1 && NTriplesUtil.isLetterOrNumber(c)) {
+		while (c != -1 && NTriplesUtil.isValidCharacterForBNodeLabel(c)) {
+			if(NTriplesUtil.isDot(c) && !NTriplesUtil.isValidCharacterForBNodeLabel(peekCodePoint())) {
```

Comment:
Shouldn't you also do an `unread` here on the dot char? Unless I misunderstand, in this situation, we should not consider the dot part of the node id, so it should be put back on the reader to be processed outside the context of the nodeId, I think.

Reply:
Ah never mind, of course not, the last-read char is returned by the method.